File: app/src/main/java/com/leaguechampions/ui/settings/SettingsComponent.java
package com.leaguechampions.ui.settings;

import com.leaguechampions.injection.AppComponent;
import com.leaguechampions.injection.scope.ActivityScoped;

import dagger.Component;

@ActivityScoped
@Component(dependencies = {AppComponent.class}, modules = {SettingsPresenterModule.class})
public interface SettingsComponent {
    void inject(SettingsActivity activity);
}

File: app/src/main/java/com/leaguechampions/ui/settings/SettingsComponent.kt
package com.leaguechampions.ui.settings

import com.leaguechampions.injection.AppComponent
import com.leaguechampions.injection.scope.ActivityScoped

import dagger.Component

@ActivityScoped
@Component(dependencies = [(AppComponent::class)], modules = [(SettingsPresenterModule::class)])
interface SettingsComponent {
    fun inject(activity: SettingsActivity)
}
